app/src/main/java/ca/llamabagel/transpo/ui/trips/TripsFragment.kt

```diff
@@ -22,6 +49,83 @@ class TripsFragment : Fragment() {
         return inflater.inflate(R.layout.trips_fragment, container, false)
     }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        viewModel.viewerData.observe(this, Observer {
+            view.findViewById<RecyclerView>(R.id.recycler_view).apply {
+                adapter = TripsAdapter(it)
+            }
+        })
+
+        val mapView = view.findViewById<MapView>(R.id.map_view)
+        mapView.onCreate(savedInstanceState)
+        mapView.getMapAsync { map ->
+            map.setStyle(Style.TRAFFIC_DAY) {
+                prepareMap(map)
+            }
+        }
+    }
+
+    override fun onStart() {
+        super.onStart()
+        view?.findViewById<MapView>(R.id.map_view)?.onStart()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        view?.findViewById<MapView>(R.id.map_view)?.onResume()
+    }
+
+    override fun onStop() {
+        super.onStop()
+        view?.findViewById<MapView>(R.id.map_view)?.onStop()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        view?.findViewById<MapView>(R.id.map_view)?.onDestroy()
+        viewModel.clearSelection()
+    }
+
+    override fun onLowMemory() {
+        super.onLowMemory()
+        view?.findViewById<MapView>(R.id.map_view)?.onLowMemory()
+    }
+
+    private fun prepareMap(map: MapboxMap) {
+        // TODO: Move this into a domain layer or something like damn
+        viewModel.viewerData.observe(this, Observer { items ->
+            map.style?.addImage(
+                "marker-icon-id",
+                BitmapFactory.decodeResource(requireActivity().resources, R.drawable.mapbox_marker_icon_default)
+            )
+
+            val points = items.asSequence()
+                .filter { item -> item is TripItem }
+                .map { item -> item as TripItem }
+                .filter { trip -> trip.tripUiModel.trip.adjustmentAge > 0 }
+                .map { trip -> Point.fromLngLat(trip.tripUiModel.trip.longitude!!, trip.tripUiModel.trip.latitude!!) }
+                .toList()
+
+            val geometry = Feature.fromGeometry(MultiPoint.fromLngLats(points))
+            if (points.size > 1) {
+                val lngLatBounds =
+                    LatLngBounds.Builder().includes(points.map { LatLng(it.latitude(), it.longitude()) }).build()
+                map.moveCamera(CameraUpdateFactory.newLatLngBounds(lngLatBounds, 32))
+            } else if (points.isNotEmpty()) {
+                map.moveCamera(CameraUpdateFactory.newLatLngZoom(LatLng(points[0].latitude(), points[0].longitude()), 15.0))
```

Comment:
Exceeds 80 char limit. Also we shouldn't have magic numbers

Reply:
tEmpOrARy

...but ok